spec/views/users/index.html.erb_spec.rb

```diff
@@ -1,9 +1,51 @@
 require 'rails_helper'
+RSpec.describe 'user_index_path', type: :system do
+  describe 'User_index_view' do
+    before do
+      @users = [
+        User.create(id: 444, name: 'Nicole', photo: '#photo_nicole', bio: 'bio', posts_counter: 0),
+        User.create(name: 'Tom', photo: '#photo_tom', bio: 'bio', posts_counter: 19),
+        User.create(name: 'Candy', photo: '#photo_candy3', bio: 'bio', posts_counter: 0),
+        User.create(name: 'Nicole', photo: '#photo_nicole6', bio: 'bio', posts_counter: 0),
+        User.create(name: 'Lary', bio: 'bio', posts_counter: 0)
+      ]
 
-RSpec.describe 'users/index.html.erb', type: :view do
-  it 'displays a list of users' do
-    render
-    expect(rendered).to include('<h1>List of Users</h1>')
-    expect(rendered).to include('<p>Find me in app/views/users/index.html.erb</p>')
+      visit users_path
+    end
+
+    after(:each) do
+      User.destroy_all
+    end
+
+    it 'I can see the username of all other users.' do
+      @users.each do |u|
+        expect(page).to have_content u.name
+      end
+    end
+
+    it 'I can see the user\'s profile picture', retry: 3 do
+      @users.each do |user|
+        within(".user-#{user.id} .photo-box") do
+          if user.photo.present?
+            expect(page).to have_css(".user-photo[src='#{user.photo}']", visible: :all)
+          else
+            # If there's no photo, check that the placeholder-box is present
+            expect(page).to have_css('.placeholder-box', visible: :all)
+          end
+        end
+      end
+    end
+
+    it 'I can see the number of posts each user has written.' do
+      @users.each do |u|
+        expect(page).to have_content('Number of Posts:')
+        expect(page).to have_content u.posts_counter
+      end
+    end
```

Comment:
- [ ] There is one test case is failing, Please fix that and ensure that all tests are passing.

Reply:
- [ ] As mentioned by the previous reviewer the test to see the number of posts each user has written still failing. Kindly check it and make sure all the test are passing.
![Screenshot from 2023-12-19 21-31-20](https://github.com/richie1988/blog-app/assets/91142432/4d62f419-0032-41df-b696-948392a89586)